### ISSUE TITLE
classic mod notelock dropped slider behavior 

### ISSUE THREAD

**Issue by Consuderably** [ppy/osu#27270]
### Type

Game behaviour

### Bug description

the classic mod notelock implementation might need some tweaks concerning dropped slider ends/stacked sliders

### Screenshots or videos

https://github.com/ppy/osu/assets/102204450/c7542e7a-cb65-45ac-a5be-d8712c04e294

### Version

Examples recorded on development build 
easily replicated with clean install on latest release of Lazer 2024.219.0

### Logs

not necessary

**Comment by bdach** [comment]
This isn't a "notelock" thing, it's a "2B" thing. See: https://github.com/ppy/osu/issues/5539.

**Comment by Consuderably** [comment]
I realize I wasn't very descriptive I and made an assumption that the slider chainmiss judgement as a direct result of classic notelock being mod setting enabled and didn't provide a reference point but why exactly isn't it a "notelock" thing

https://github.com/ppy/osu/assets/102204450/de44ea11-4fd1-4af8-9f69-1c68b9ae2d63

**Comment by respektive** [comment]
I don't think this is the same issue? The other one seems to be just about the notes being visually behind the sliders instead of in front.

This on is indeed an issue with the classic mod notelock implementation. Playing the map shown in the video without any mods works as intended, but when enabling classic mod with notelock enabled none of the stacked sliders can be hit except the first one. This also breaks the stable replays on the leaderboard.

Map link: https://osu.ppy.sh/beatmapsets/756794#osu/1605148

**Comment by bdach** [comment]
Unless this can be demonstrated to break a non-2B beatmap I don't foresee this being addressed quickly.

**Comment by Karoo13** [comment]
@bdach from #24762 "Now sliders become hit only after they've been fully... slid"
this is not how it works in stable, and is causing the issue reported here. the stable behavior prevents clicking something behind a sliderhead if the slider is not fully slid AND the sliderhead's 50 window has not expired.

the correct statement would be "sliderheads become hit only after they've fully slid OR their 50 window expires". this actually implies another discrepancy, in stable you can not click a sliderhead after it has finished sliding because it is already hit (and counted as a miss). this side of the behavior is better captured as a separate concept from notelock though, i would categorize it as removal of retroactive slider following (described here https://github.com/ppy/osu/issues/32359#issuecomment-2951641183)
